cli/actions/run_test_action.go

```diff
@@ -54,28 +57,96 @@ func (a runTestAction) Run(ctx context.Context, args RunTestConfig) error {
 		return fmt.Errorf("--junit option requires --wait-for-result")
 	}
 
-	metadata := a.getMetadata()
 	a.logger.Debug(
 		"Running test from definition",
 		zap.String("definitionFile", args.DefinitionFile),
+		zap.String("environment", args.EnvID),
 		zap.Bool("waitForResults", args.WaitForResult),
 		zap.String("junit", args.JUnit),
 	)
+
+	envID, err := a.processEnv(ctx, args.EnvID)
+	if err != nil {
+		return fmt.Errorf("could not run definition: %w", err)
+	}
+
 	params := runDefParams{
 		DefinitionFile: args.DefinitionFile,
+		EnvID:          envID,
 		WaitForResult:  args.WaitForResult,
 		JunitFile:      args.JUnit,
-		Metadata:       metadata,
+		Metadata:       a.getMetadata(),
 	}
 
-	err := a.runDefinition(ctx, params)
+	err = a.runDefinition(ctx, params)
 	if err != nil {
 		return fmt.Errorf("could not run definition: %w", err)
 	}
 
 	return nil
 }
 
+func stringReferencesFile(path string) bool {
+	return strings.HasPrefix(path, "./")
+}
+
+func (a runTestAction) processEnv(ctx context.Context, envID string) (string, error) {
+	if !stringReferencesFile(envID) { //not a file, do nothing
+		return envID, nil
+	}
+
+	envVars, err := godotenv.Read(envID)
```

Comment:
I'd expect the environment file to be in the following format and not an .env file:

```yaml
---
type: Environment
id: something
name: something
description: an env
values:
  - key: HOST
    value: localhost
  - key: PORT
    value: 1245